package-res/resources/web/pentaho/data/CellTuple.js

```diff
@@ -26,25 +26,29 @@ define([
       this.base();
 
       // Fill with null valued cells.
-      var i = this.length, SL = this.structure.length;
+      var i = this.length;
+      var SL = this.structure.length;
       while(i++ < SL) this.add(null);
     },
 
-    //region IListElement
+    // region IListElement
     elemName: "CellTuple",
-    //endregion
+    // endregion
 
-    //region IWithKey implementation
+    // region IWithKey implementation
     keyName:  "key",
 
     _key: null,
 
     get key() {
       var key = this._key;
-      if(key == null) key = this._key = this.map(function(cell) { return cell.key; }).join("~");
+      if(key == null)
```

Comment:
There's a rule about the maximum number of statements in a single line. I fixed that one, but should have added parenses as well. Anyway, I wouldn't go as far as qualifying it as "much more _dangerous_"...
Anyway, my goal with this PR is not to discuss the current rules — which would take time we don't currently have — but simply to configure the current rules in an ESLint file.

I deactivated the parens rule; currently, ESLint has no configuration that supports the current style. Interestingly, there is an active PR to add it to ESLint: https://github.com/eslint/eslint/issues/6141 (`"multi-or-nest"` plus not enforcing unncessary parens are not used).